.github/workflows/test.yml

```diff
@@ -0,0 +1,89 @@
+name: Test
+
+on: push
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup node and cache yarn dependencies
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.15.0'
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Run linter
+        run: yarn lint
+
+  test-ui:
+    needs: [linter]
```

Comment:
Why do we need the `linter` step here?
Should we move linter into a dedicated workflow instead? 🤔 
Then we would have `test.yml` and `lint.yml`

Reply:
Make it in separate https://github.com/Lahphim/react-surveyor-challenge/pull/47/commits/4bf7bdaede665c34b413afd637dcb249da7f3505